Content.Server/GameObjects/Components/Kitchen/ReagentGrinderComponent.cs

```diff
@@ -0,0 +1,368 @@
+                foreach (var entity in _chamber.ContainedEntities)
+                {
+                    if (!canJuice && entity.HasComponent<JuiceableComponent>()) canJuice = true;
+                    if (!canGrind && entity.HasComponent<GrindableComponent>()) canGrind = true;
+                    if (canJuice && canGrind) break;
+                }
+
+                UserInterface?.SetState(new ReagentGrinderInterfaceState
+                (
+                    _busy,
+                    HasBeaker,
+                    canJuice,
+                    canGrind,
+                    _chamber.ContainedEntities.Select(item => item.Uid).ToArray(),
+                    //Remember the beaker can be null!
+                    _heldBeaker?.Solution.Contents.ToArray()
+                ));
+                _uiDirty = false;
+            }
+        }
+
+        private void EjectSolid(EntityUid entityID)
+        {
+            if (_busy)
+                return;
+
+            if (Owner.EntityManager.EntityExists(entityID))
+            {
+                var entity = Owner.EntityManager.GetEntity(entityID);
+                _chamber.Remove(entity);
+
+                //Give the ejected entity a tiny bit of offset so each one is apparent in case of a big stack,
+                //but (hopefully) not enough to clip it through a solid (wall).
+                const float ejectOffset = 0.4f;
+                entity.Transform.LocalPosition += (_random.NextFloat() * ejectOffset, _random.NextFloat() * ejectOffset);
+            }
+            _uiDirty = true;
+        }
+
+        /// <summary>
+        /// If this component contains an entity with a <see cref="SolutionContainerComponent"/>, eject it.
+        /// Tries to eject into user's hands first, then ejects onto dispenser if both hands are full.
+        /// </summary>
+        private void EjectBeaker(IEntity user)
+        {
+            if (!HasBeaker || _busy)
+                return;
+
+            //Eject the beaker into the hands of the user.
+            _beakerContainer.Remove(_beakerContainer.ContainedEntity);
+
+            //UpdateUserInterface();
+
+            if (!user.TryGetComponent<HandsComponent>(out var hands) || !_heldBeaker!.Owner.TryGetComponent<ItemComponent>(out var item))
+                return;
+            if (hands.CanPutInHand(item))
+                hands.PutInHand(item);
+
+            _heldBeaker = null;
+            _uiDirty = true;
+            SetAppearance(ReagentGrinderVisualState.NoBeaker);
+        }
+
+        void IActivate.Activate(ActivateEventArgs eventArgs)
+        {
+            if (!eventArgs.User.TryGetComponent(out IActorComponent? actor) || !Powered)
+            {
+                return;
+            }
+            _uiDirty = true;
+            UserInterface?.Toggle(actor.playerSession);
+        }
+
+        public async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+            if (!Powered)
```

Comment:
You should be able to put items in the grinder if its unpowered, just not turn it on.